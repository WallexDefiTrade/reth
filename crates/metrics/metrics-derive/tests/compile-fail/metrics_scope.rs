extern crate reth_metrics_derive;
use reth_metrics_derive::Metrics;

fn main() {}

#[derive(Metrics)]
#[metrics()]
struct CustomMetrics;

#[derive(Metrics)]
#[metrics(scope = value)]
struct CustomMetrics2;

#[derive(Metrics)]
#[metrics(scope = 123)]
struct CustomMetrics3;

#[derive(Metrics)]
#[metrics(scope = "some.scope", scope = "another.scope")]
struct CustomMetrics4;

#[derive(Metrics)]
#[metrics(random = "value")]
struct CustomMetrics5;
