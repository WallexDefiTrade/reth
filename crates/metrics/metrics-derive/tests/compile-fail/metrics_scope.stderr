error: `scope = ..` must be set.
 --> tests/compile-fail/metrics_scope.rs:7:1
  |
7 | #[metrics()]
  | ^^^^^^^^^^^^

error: expected literal
  --> tests/compile-fail/metrics_scope.rs:11:19
   |
11 | #[metrics(scope = value)]
   |                   ^^^^^

error: Value **must** be a string literal.
  --> tests/compile-fail/metrics_scope.rs:15:19
   |
15 | #[metrics(scope = 123)]
   |                   ^^^

error: Only single `scope = ..` value is supported.
  --> tests/compile-fail/metrics_scope.rs:19:11
   |
19 | #[metrics(scope = "some.scope", scope = "another.scope")]
   |           ^^^^^^^^^^^^^^^^^^^^

error: Only single `scope = ..` value is supported.
  --> tests/compile-fail/metrics_scope.rs:23:11
   |
23 | #[metrics(random = "value")]
   |           ^^^^^^^^^^^^^^^^
